### DISCUSSION TITLE
Ideas para Javascript

### DISCUSSION THREAD

**Discussion by rpgrca** [rpgrca/proyecto-ispc-fullstack#155] [Ideas]
Algunas cosas muy simples que se pueden hacer en Javascript. Si se les ocurren otras posteenlas y tratamos de ver cuáles son las que puede realizar cada uno.

- En la página de login, si el usuario o el password está vacío y se presiona el botón que aparezca un mensaje diciendo que los datos son incorrectos.
- En la página de login cuando el usuario y password están completos que aparezca un mensaje diciendo que el usuario se logueó.
- En la página de registración si algún dato obligatorio está vacío y se presiona el botón que aparezca un mensaje diciendo que los datos son incorrectos.
- En la página de registración si todos los datos están correctos que aparezca un mensaje diciendo que el usuario se registró correctamente.
- En la página de perfil si se intenta borrar algún dato que tiene que estar disponible (por ejemplo se intenta dejar el nombre vacío) que aparezca un mensaje diciendo que los datos son incorrectos.
- En la página de perfil si el password está vacío cuando se quiere cambiar el password que diga que deb e ingresar el password existente.
- En la página de perfil el password nuevo y la repetición del password nuevo no son iguales que aparezca un mensaje diciendo que deben ser iguales.
- En la página de perfil si todos los datos están correctos que diga que el nuevo perfil se grabó.
- En la página de subasta de cliente cuando se presiona alguno de los botones para subir la apuesta que aparezca un mensaje diciendo que la nueva apuesta fue aceptada.
- En la página de subasta del martillero cuando se presiona el botón para cerrar la subasta debe aparecer un mensaje diciendo que el ganador del lote fue tal persona.

Hay muchas cosas más que se podrían hacer mostrando message boxes, sin embargo me imagino que van a querer que toquemos el DOM y que modifiquemos la página (si no es con JQuery a mano). A medida que vayan aprendiendo Javascript propondría que se animen a meter esos mensajes como para ir ganando tiempo mientras pensamos en cosas más complicadas para implementar.

**Comment by Brian-Almada** [comment]
Sí, genial.
Para aportar, sugiero que nos animemos con el botón del darkmode (más que por estilo, para tener una práctica de código y un buen desafío). Luego, para la parte de catálogo en donde están los productos creo que es bueno tener un botón que aparezca cuando scrolleamos mucho para subir de un toque al inicio y no tener que subir a mano con la barra de scroll (o prendiendo fuego el dedo gordo en el celu, jeje). Con JS se pueden manejar los loaders para que la espera de las respuestas estén animadas y luego aparezcan los cartelitos (se usa css para la animación y con js el setTimeOut).  Algo un poco más jugado, si hay tiempo, es hacer un filtro de búsqueda en los artículos (que pueden estar ordenados por fecha de subasta, por alguna categoría, por precio, etc..). Para el martillero que va a tener una cam web (supongo) se puede manejar el error para que le aparezca cuando no puede establecer la conexión con la cam y hacerlo con un cartelito que avise qué tipo de error (pj: NotFoundError: Requested device not found). Y creo que no puede faltar la cuenta regresiva a la subasta, que aparezca arriba en algún lugar mientras se navega, eso estaría vistoso, je.

Pongo lo que se me ocurrió en lista para mejor lectura:
-Botón darkmode 
-Botón scrollTop
-Loaders
-Filtro de búsqueda
-Verificar conexión de camweb
-Cuenta regresiva

**Comment by AhumadaCeleste** [comment]
Aporto ideas a medida de que voy comprendiendo la funcionalidad de Javascript a modo de que podamos comenzar a crear la issues.

-En creación de lote aparecer un cartel anunciando error en caso de que falte asignar: ID lote, Precio base.
-Que a los participantes de la puja se le indique que la puja se encuentra finalizada al cerrar el proceso.
-Durante de subasta pasaría alguna publicidad de almoneda.
-Colocarle animación a algunos texto con efecto de aparición.
- Crear un Popup que indique que VIDON Certifico normas ISO al ingresar al sitio.

**Comment by NataliaAlvarezIspc** [comment]
Me ocupo de esta:
En la página de subasta de cliente cuando se presiona alguno de los botones para subir la apuesta que aparezca un mensaje diciendo que la nueva apuesta fue aceptada.
